src/troute_model.py

```diff
@@ -231,8 +264,28 @@ def run(self, values: dict, until=300):
         
         # update model time
         self._time += self._time_step * nts
+        
+        if self.showtiming:
+            output_end_time = time.time()                           # end of recording time for creating output #
+            self.task_times['output_time'] += output_end_time - run_end_time
 
-
+    def finalize(self,):
+            if self.showtiming:
+                print('***************** TIMING SUMMARY *****************')
+                print('----------------------------------------')
+                total_time = (round(self.task_times['network_time'], 2) + 
+                              round(self.task_times['data_assimilation_time'], 2) + 
+                              round(self.task_times['forcing_time'], 2) + 
+                              round(self.task_times['run_time'], 2) + 
+                              round(self.task_times['output_time'], 2)
+                              )
+                for key in self.task_times:
+                    time_value = self.task_times[key]
+                    percentage = (time_value / total_time) * 100
+                    print(
+                    f'{key} construction: {time_value:.2f} secs, {percentage:.2f} %'
```

Comment:
Can you update the print statement with specific text rather than using the task_times.keys()? Take a look at the print statements in __main__.py, but something like "Network graph construction: ..." rather than "network_time construction: ..."